kobo/apps/audit_log/tests/test_signals.py

```diff
@@ -138,3 +152,95 @@ def test_loginas(self):
         audit_log = AuditLog.objects.first()
         self.assertEqual(audit_log.user.id, new_user.id)
         self.assertEqual(audit_log.action, AuditAction.AUTH)
+
+
+@ddt
+class ProjectHistoryLogsSignalsTestCase(BaseTestCase):
+
+    fixtures = ['test_data']
+
+    def setUp(self):
+        super().setUp()
+        self.wsgi_request = RequestFactory().post('/')
+        self.asset = Asset.objects.get(pk=1)
+        # 'someuser' owns the asset, so use 'anotheruser' for testing perm assignments
+        self.user = User.objects.get(username='anotheruser')
+        request_patcher = patch(
+            'kobo.apps.audit_log.signals.get_current_request',
+            return_value=self.wsgi_request,
+        )
+
+        request_patcher.start()
+        self.addCleanup(request_patcher.stop)
+
+    @data(
+        post_assign_perm,
+        post_remove_perm,
+        post_assign_partial_perm,
+        post_remove_partial_perms,
+    )
+    def test_receivers_add_necessary_fields_to_request(self, signal):
+        # use all possible parameters for each signal
+        # so we can re-use the test
+        signal.send(
+            sender=Asset,
+            instance=self.asset,
+            user=self.user,
+            codename=PERM_VIEW_ASSET,
```

Comment:
Should it  be `PERM_PARTIAL_SUBMISSIONS`? or you did it in purpose because of your test? 
If it is the latter, you add a comment because `PERM_VIEW_ASSET` should never be added with partial perms.

Reply:
In this case it doesn't matter since I'm just testing the signal, not actually adding permissions to anything.